CIE-5497 immutable CR fields email and company name

bbyreddy opened the pull request.

bbyreddy committed:
Updated doc for statig that the email and company are immutable.

bbyreddy committed:
Updated the edit user doc link.

bbyreddy committed:
Minor changes to the links.

bbyreddy committed:
More updates to the updating after installation.

BeateRixen commented on content/edge-kubernetes/manage-edge-bundle/modify-edge.md:
```diff
@@ -10,7 +10,7 @@ In these examples, we assume that the Edge object is called `c8yedge` and is in
 ### Basic post-installation configuration {#basic-post-installation-configuration}
 
-Immediately post-installation, you will probably want to configure an SSL certificate for your Edge, and give it a custom domain.
+Immediately after installing Edge using the `c8yedge` tool, you will probably want to configure an SSL certificate for your Edge, and give it a custom domain. And also set the Edge tenant's name (Company name) and the admin user email.
```
```suggestion
After installing Edge using the `c8yedge` tool, you might want to configure an SSL certificate for your Edge, give it a custom domain, and set the Edge tenant's name (company name) and the admin user email.
```

BeateRixen commented on content/edge-kubernetes/manage-edge-bundle/modify-edge.md:
````diff
@@ -19,13 +19,19 @@ kubectl create secret tls edge-tls-secret --namespace=c8yedge \
 ```
 
-Then apply the configuration:
+Then apply the below configuration for configuring and SSL certificate and a custom domain:
````
```suggestion
Then apply the below configuration for configuring an SSL certificate and a custom domain:
```

bbyreddy committed:
Apply suggestions from code review

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy committed:
Merge branch 'release/y2025' into CIE-5497/immutable-email-and-company-name

bbyreddy committed:
Release notes for CIE-5497

bbyreddy committed:
Release notes for CIE-5497

BeateRixen commented on content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md:
```diff
@@ -0,0 +1,24 @@
+---
+date: 2024-01-09
+title: Edge CR fields `company` and `email` are now immutable
```
```suggestion
title: Edge CR fields "company" and "email" are now immutable
```

BeateRixen replied:
Code formatting looks bad in the title.

BeateRixen commented on content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md:
```diff
@@ -0,0 +1,24 @@
+In earlier releases, updating the `company` or `email` fields in the Edge Custom Resource (CR) after installation was technically allowed, but these updates were not reflected in the running Edge instance.
+
+Starting with this release:
+  - If the Edge was initially installed using the `c8yedge` tool, one subsequent update to these fields is permitted.
```
```suggestion
  - If the Edge was initially installed using the `c8yedge` tool, a subsequent update to these fields is permitted.
```

BeateRixen commented on content/edge-kubernetes/manage-edge-bundle/modify-edge.md:
```diff
@@ -10,7 +10,7 @@ In these examples, we assume that the Edge object is called `c8yedge` and is in
 ### Basic post-installation configuration {#basic-post-installation-configuration}
 
-Immediately post-installation, you will probably want to configure an SSL certificate for your Edge, and give it a custom domain.
+Immediately after installing Edge using the `c8yedge` tool, you will probably want to configure an SSL certificate for your Edge, and give it a custom domain. And also set the Edge tenant's name (Company name) and the admin user email.
```
Probably you did not accept my suggestions for good reasons. But at least the last sentence shouldn't start with "And". I would remove it.

bbyreddy replied:
Sorry, I didn't realize that I haven't accepted your suggestion. I might have added your suggestion to a batch and didn't commit it. Committed your suggestion now :)

bbyreddy committed:
Update content/edge-kubernetes/manage-edge-bundle/modify-edge.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy committed:
Update content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy committed:
Update content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

rthrippleton commented on content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md:
```diff
@@ -0,0 +1,24 @@
+version: 2025.0.8
+---
+
+In earlier releases, updating the `company` or `email` fields in the Edge Custom Resource (CR) after installation was technically allowed, but these updates were not reflected in the running Edge instance.
```
Pedantically ... I think `email` was already immutable. `[spec.email] cannot be modified for an existing installation. To update the admin user details, refer to https://cumulocity.com/docs/standard-tenant/managing-users/#to-edit-a-user` got spat out in previous builds to 2025.0.8.

But of course we didn't document it before, so I'm happy to be technically incorrect here if you are.

bbyreddy replied:
Yeah, I understand that the updating email wasn't possible earlier. I kept both for completeness. I will update the statement a bit to keep it much closer to the reality.

bbyreddy replied:
@rthrippleton, Please check the updated change-log summary https://github.com/Cumulocity-IoT/c8y-docs/pull/3865/commits/d2e44df53d20de92ec6d501d8214f63f359a51c6

bbyreddy committed:
Update changelog for Edge version 2025.0.8

Clarified the immutability validation for 'company' and 'email' fields in the Edge Custom Resource after installation.

rthrippleton commented on content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md:
```diff
@@ -0,0 +1,24 @@
+
+Starting with this release, the `company` and the `email` fields are now validated for immutability after installation:
+  - If the Edge was initially installed using the `c8yedge` tool, a subsequent update to these fields is permitted.
+  - Any further modifications to these fields after the initial setup will trigger a validation error.
```
It's not an 'update' when it wasn't set in the first place, pedantic word-smithing:
```suggestion
Starting with this release, the `company` and the `email` fields are now validated for immutability after installation.

If the Edge was initially installed using the `c8yedge` tool, subsequently setting these fields is permitted once and once only.
```

rthrippleton reviewed (commented):
Can you also make sure that the formatting for "c8yedge" is brought in line with https://github.com/Cumulocity-IoT/c8y-docs/pull/3769 throughout this PR?

bbyreddy committed:
Update content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md

Co-authored-by: Richard Thrippleton <richard.thrippleton@softwareag.com>

bbyreddy committed:
Updated the styling for "c8yedge" in the c8yedge tool

BeateRixen commented on content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md:
```diff
@@ -0,0 +1,24 @@
+
+Starting with this release, the `company` and the `email` fields are now validated for immutability after installation.
+
+If the Edge was initially installed using the **c8yedge** tool, subsequently setting these fields is permitted once and once only.
```
```suggestion
If Edge was initially installed using the **c8yedge** tool, subsequently setting these fields is permitted once and once only.
```

BeateRixen replied:
Not sure what "once and once only" means here?

bbyreddy replied:
These settings are allowed to be changed via the Edge Custom Resource only once. Any subsequent changes to these fields can only be changed via UI or the REST API.

bbyreddy replied:
Addressed in https://github.com/Cumulocity-IoT/c8y-docs/pull/3865/commits/e3bb9b39d09b429b90f1ad5a05b1f1ab381309cd

bbyreddy committed:
Update content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy committed:
Merge branch 'release/y2025' into CIE-5497/immutable-email-and-company-name

BeateRixen commented on content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md:
```diff
@@ -0,0 +1,24 @@
+
+Starting with this release, the `company` and the `email` fields are now validated for immutability after installation.
+
+If Edge was initially installed using the **c8yedge** tool, subsequently setting these fields is permitted once and once only.
```
```suggestion
If Edge was initially installed using the **c8yedge** tool, subsequently setting these fields is permitted only once.
```

bbyreddy replied:
Addressed in https://github.com/Cumulocity-IoT/c8y-docs/pull/3865/commits/e3bb9b39d09b429b90f1ad5a05b1f1ab381309cd

BeateRixen commented on content/change-logs/edge/edge-2025.0.8-company-and-email-immutable.md:
```diff
@@ -0,0 +1,24 @@
+version: 2025.0.8
+---
+
+In earlier releases, the `company` field in the Edge Custom Resource (CR) could be modified after installation. However, the change was not reflected in the Edge tenant details.
```
I would actually remove all, or at least 2 of the empty rows here. Usually,  in release notes entries, we do not use so many paragraphs.

bbyreddy replied:
Addressed in https://github.com/Cumulocity-IoT/c8y-docs/pull/3865/commits/e3bb9b39d09b429b90f1ad5a05b1f1ab381309cd

bbyreddy committed:
Handled the review comments.

rthrippleton committed:
Merge branch 'release/y2025' into CIE-5497/immutable-email-and-company-name
